In [1]:
import numpy as np
from IPython.display import HTML
from bokeh.plotting import output_notebook, show
import genomes_dnj.lct_interval.series_plots as dm
import genomes_dnj.lct_interval.anal_series as an
import genomes_dnj.lct_interval.series_masks as sm
output_notebook(hide_banner=True)

<h3>123_1561 Selectors</h3>
<div style="width:700px">
<p>
The series 123_1561 extends across a 435,000 base region of chromosome 2
that includes the gene zranb3 and about half of the gene r3hdm1.  This range
is contained by 117_1685's and located towards its upper end.
<p>
The processes that selected 123_1561 are similar to those that selected 117_1685.
The three processes that generated 193_843, 67_329, and 209_56 all selected
123_1561, 117_1685 and 62_1265.  Most of the chromosome samples that express
either 193_843 or 67_329 express all 123 of the series 123_1561 SNPs.
None of the 56 chromosome samples expressing 209_56 express all 123 SNPs.  But, most
of them do express 122 of those SNPs.
<p>
Two series, 180_251 and 22_35 select 123_1561 without 117_1685.  Most of the
chromosome samples expressing both series express all 123 of the series 123_1561 SNPs
All of those samples express some significant size fragment of 117_1685.
Both of these hierarchies include the series 13_1696 in association with
123_1561.  The 251 chromosome samples that express 180_251 are a large part of the
434 samples that express the series 28_434.  The 35 samples that express
22_35 do not express 28_434.
<p>
Another 25 chromosome samples express an association of series that includes 28_434,
25_27, 123_1561, and 117_1685.  These samples only express 115 or 116 of the
series 123_1561 SNPs.
</div>
<h3>193_843</h3>
<div style="width:700px">
<p>
The first plot in this notebook shows data for the chromosome samples that express the 
series 193_843.  All 843 samples express 123_1561.  SNP counts show that 
729 of those samples express all 123 series 123_1561 SNPs.
Another 104 express 122 SNPs.
</div>

In [2]:
plt_obj = dm.superset_yes_no([dm.di_123_1561, dm.di_193_843], min_match=0.5)
plt = plt_obj.do_plot()
am_193_843 = plt_obj.plot_context.yes_allele_mask
show(plt)

In [3]:
HTML(plt_obj.get_html())

In [4]:
count_data = an.sa_123_1561.unique_snps_per_allele(am_193_843)
count_data

array([(113,   3), (116,   1), (120,   1), (121,   5), (122, 104),
       (123, 729)], 
      dtype=[('count', '<u2'), ('snps', '<u2')])

<h3>67_329</h3>
<div style="width:700px">
<p>
This plot shows data for 325 of the 329 series 67_329 chromosome samples that also express the series 
123_1561.  SNP counts show that 262 of those samples express all 123 series 123_1561 SNPs.
Anoter 42 express 122 of the SNPs.
</div>

In [5]:
plt_obj = dm.superset_yes_no([dm.di_123_1561, dm.di_67_329], min_match=0.5)
plt = plt_obj.do_plot()
am_67_329 = plt_obj.plot_context.yes_allele_mask
show(plt)

In [6]:
HTML(plt_obj.get_html())

In [7]:
count_data = an.sa_123_1561.unique_snps_per_allele(am_67_329)
count_data

array([(117,  11), (120,   2), (121,   8), (122,  42), (123, 262)], 
      dtype=[('count', '<u2'), ('snps', '<u2')])

<h3>180_251</h3>
<div style="width:700px">
<p>
This plot shows data for the 251 chromosome samples that express the 180 SNP series
180_251.  All 251 of these samples express the series 123_1561.
SNP counts show that 159 of the samples express all 123 of
the series 123_1561 SNPs.  Another 76 samples express 122 of them.
None of the chromosome samples expresses the series 117_1685.  But all
of the samples do express significant size fragments of 117_1685.
This hierarchy is the most significant one for establishing 123_1561
as a separate series distinct from 117_1685 and 62_1265.
</div>

In [8]:
plt_obj = dm.superset_yes_no([dm.di_123_1561, dm.di_180_251], min_match=0.5)
plt = plt_obj.do_plot()
am_180_251 = plt_obj.plot_context.yes_allele_mask
show(plt)

In [9]:
HTML(plt_obj.get_html())

In [10]:
count_data = an.sa_123_1561.unique_snps_per_allele(am_180_251)
count_data

array([(120,   7), (121,   9), (122,  76), (123, 159)], 
      dtype=[('count', '<u2'), ('snps', '<u2')])

In [11]:
count_data = an.sa_117_1685.unique_snps_per_allele(am_180_251)
count_data

array([(59,   1), (60,  27), (61, 164), (62,  50), (63,   2), (68,   1),
       (92,   4), (93,   1), (98,   1)], 
      dtype=[('count', '<u2'), ('snps', '<u2')])

<h3>209_56</h3>
<div style="width:700px">
<p>
This plot shows data for the 56 chromosome samples that express the 
209 SNP series 209_56.  All 56 of the samples express
the series 123_1561.  SNP counts show that none of those 56
samples express all 123 SNPs.  But 49 of the samples
do express 122 of the 123_1561 SNPs.  The other 7 samples
express 121 of them.
</div>

In [12]:
plt_obj = dm.superset_yes_no([dm.di_123_1561, dm.di_209_56], min_match=0.1)
plt = plt_obj.do_plot()
am_209_56 = plt_obj.plot_context.yes_allele_mask
show(plt)

In [13]:
HTML(plt_obj.get_html())

In [14]:
count_data = an.sa_123_1561.unique_snps_per_allele(am_209_56)
count_data

array([(121,  7), (122, 49)], 
      dtype=[('count', '<u2'), ('snps', '<u2')])

<h3>22_35</h3>
<div style="width:700px">
<p>
This plot shows data for the chromosome samples that express the series
22_35.  All 35 of the samples express the series 123_1561.
SNP counts show that 26 of the samples express all 123 series
123_1561 SNP's.  Another 5 express 122 of those SNPs.
</div>

In [15]:
plt_obj = dm.superset_yes_no([dm.di_123_1561, dm.di_22_35], min_match=0.5)
plt = plt_obj.do_plot()
am_22_35 = plt_obj.plot_context.yes_allele_mask
show(plt)

In [16]:
HTML(plt_obj.get_html())

In [17]:
count_data = an.sa_123_1561.unique_snps_per_allele(am_22_35)
count_data

array([(119,  1), (121,  3), (122,  5), (123, 26)], 
      dtype=[('count', '<u2'), ('snps', '<u2')])

In [18]:
count_data = an.sa_117_1685.unique_snps_per_allele(am_22_35)
count_data

array([(43, 22), (44,  1), (46,  1), (47,  1), (53,  1), (58,  1), (59,  8)], 
      dtype=[('count', '<u2'), ('snps', '<u2')])

<h3>25_27</h3>
<div style="width:700px">
<p>
This plot shows data for the 25 series 25_27 chromosome samples that also
express the series 123_1561.  All 25 samples also express the series
117_1685.  SNP counts show that 23 of the samples only express a
115 SNP fragment of the 123 SNP series 123_1561 SNPs.  The other 2 express
a 116 SNP fragment.
</div>

In [19]:
plt_obj = dm.superset_yes_no([dm.di_123_1561, dm.di_25_27], min_match=0.5)
plt = plt_obj.do_plot()
am_25_27 = plt_obj.plot_context.yes_allele_mask
show(plt)

In [20]:
HTML(plt_obj.get_html())

In [21]:
count_data = an.sa_123_1561.unique_snps_per_allele(am_25_27)
count_data

array([(115, 23), (116,  2)], 
      dtype=[('count', '<u2'), ('snps', '<u2')])

<h3>Noise</h3>
<div style="width:700px">
<p>
This plot shows data for the 26 remaining chromosome samples that express
123_1561 without any indentifiable association with a selector series.
SNP counts show that 10 of the samples express all 123 of the 
123_1561 SNPs.  Another 6 express 122 of them.  All of the samples
express a signficant size fragment of 117_1685.
</div>

In [22]:
plt_obj = dm.superset_yes_no([dm.di_123_1561], 
                             [dm.di_25_27, dm.di_22_35, dm.di_209_56, dm.di_180_251, dm.di_67_329, dm.di_193_843], min_match=0.5)
plt = plt_obj.do_plot()
am_rest = plt_obj.plot_context.yes_allele_mask
show(plt)

In [23]:
HTML(plt_obj.get_html())

In [24]:
count_data = an.sa_123_1561.unique_snps_per_allele(am_rest)
count_data

array([(115,  1), (118,  1), (119,  2), (120,  3), (121,  3), (122,  6),
       (123, 10)], 
      dtype=[('count', '<u2'), ('snps', '<u2')])

In [25]:
count_data = an.sa_117_1685.unique_snps_per_allele(am_rest)
count_data

array([( 20, 1), ( 21, 1), ( 22, 2), ( 24, 2), ( 57, 1), ( 59, 2),
       ( 60, 1), ( 61, 2), ( 62, 3), (106, 1), (114, 2), (115, 2),
       (116, 2), (117, 4)], 
      dtype=[('count', '<u2'), ('snps', '<u2')])